source/class/zx/utils/AsyncMemoizer.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * This class is used to memoize return values of asynchronous functions,
+ * indxed by their input argument.
+ *
+ * Optionally, the cache can expire after a certain time.
+ *
+ * @typedef {(input: InputData) => (Promise<OutputData> | OutputData)} Generator
```

Comment:
this is not jsdoc style that we use - it looks like some kind of typescript mashup?  Wouldnt this be `@callback` (which you've used in `zx.utils.Csv` elsewhere in this PR)?